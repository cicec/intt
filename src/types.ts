export enum Bundler {
  WEBPACK = 'webpack',
  SNOWPACK = 'snowpack'
}

export enum Framework {
  REACT = 'react',
  VUE = 'vue'
}

export enum Feature {
  BABEL = 'babel',
  TYPESCRIPT = 'typescript'
}

export enum Test {
  NONE = 'none',
  JEST = 'jest',
  AVA = 'ava'
}

export type Answers = {
  name: string
  bundler?: Bundler
  framework?: Framework
  features?: Feature[]
  test?: Test
}

export type Selection =
  | 'webpack'
  | 'snowpack'
  | 'react'
  | 'vue'
  | 'babel'
  | 'typescript'
  | 'jest'
  | 'ava'

export type CLIOptions = {
  name: string
  is: Record<Selection, boolean>
}

export type Files = { [x: string]: string | Files | undefined }

export type PackageConfig = {
  scripts?: { [x: string]: string }
  dependencies?: string[]
  devDependencies?: string[]
}

export type WebpackConfig = {
  prefix?: string[]
  entry?: string
  extensions?: string[]
  rules?: any[]
  plugins?: string[]
}

export type JSONConfig = { [x: string]: any }

export type SnowpackConfig = JSONConfig

export type BabelConfig = JSONConfig

export type PluginConfig = {
  condition?: boolean
  files?: Files
  package?: PackageConfig
  webpack?: WebpackConfig
  snowpack?: SnowpackConfig
  babel?: BabelConfig
}

export type Plugin = (options: CLIOptions) => PluginConfig
