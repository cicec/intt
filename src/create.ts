import { execute, getCLIOptions, mergeDeepWith } from './utils'
import { PluginConfig, Files, Answers } from './types'

import plugins from './plugins'
import {
  generateBabel,
  generatePackage,
  generateWebpack,
  generateGitignore,
  generateSnowpack
} from './generate'

export const create = async (answers: Answers) => {
  const options = getCLIOptions(answers)
  const { name, is } = options

  const config = execute(() => {
    return plugins
      .map(plugin => plugin(options))
      .filter(({ condition }) => condition)
      .reduce<PluginConfig>((acc, elem) => {
        const isArr = (x: unknown) => Array.isArray(x)

        return mergeDeepWith((_, a, b) => (isArr(a) && isArr(b) ? [...a, ...b] : b), acc, elem)
      }, {})
  })

  const files: Files = {
    ...config.files,
    '.gitignore': generateGitignore(),
    'package.json': await generatePackage(name, config.package)
  }

  const append = (fileMap: { [x: string]: string }) => Object.assign(files, fileMap)

  if (is.babel) {
    append({ '.babelrc': generateBabel(config.babel) })
  }

  if (is.snowpack) {
    append({ 'snowpack.config.json': generateSnowpack(config.snowpack) })
  }

  if (is.webpack) {
    append({ 'webpack.config.js': generateWebpack(config.webpack) })
  }

  return files
}
