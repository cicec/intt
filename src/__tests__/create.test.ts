import { URL } from 'url'
import { create } from '../create'
import { Answers, Bundler, Feature, Framework, Test } from '../types'

const name = 'intt-app'

jest.mock('axios', () =>
  jest.fn((url: string) => {
    const { pathname } = new URL(url)
    const name = pathname.slice(1, pathname.lastIndexOf('/'))

    return Promise.resolve({ status: 200, data: { name, version: '1.0.0' } })
  })
)

test('Webpack + React + Babel', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.WEBPACK,
    framework: Framework.REACT,
    features: [Feature.BABEL]
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})

test('Webpack + React + Babel + Ts', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.WEBPACK,
    framework: Framework.REACT,
    features: [Feature.BABEL]
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})

test('Webpack + Vue', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.WEBPACK,
    framework: Framework.VUE,
    features: []
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})

test('Webpack + Vue + Babel + Ts', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.WEBPACK,
    framework: Framework.VUE,
    features: [Feature.BABEL, Feature.TYPESCRIPT]
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})

test('Snowpack + React', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.SNOWPACK,
    framework: Framework.REACT,
    features: []
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})

test('Snowpack + React + Ts', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.SNOWPACK,
    framework: Framework.REACT,
    features: [Feature.TYPESCRIPT]
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})

test('Snowpack + Vue', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.SNOWPACK,
    framework: Framework.VUE,
    features: []
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})

test('Snowpack + Vue + Ts', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.SNOWPACK,
    framework: Framework.VUE,
    features: [Feature.TYPESCRIPT]
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})

test('Webpack + React + Jest', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.WEBPACK,
    framework: Framework.REACT,
    test: Test.JEST
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})

test('Webpack + React + AVA', async () => {
  const answers: Answers = {
    name,
    bundler: Bundler.SNOWPACK,
    framework: Framework.VUE,
    test: Test.AVA
  }

  const files = await create(answers)

  return expect(files).toMatchSnapshot()
})
