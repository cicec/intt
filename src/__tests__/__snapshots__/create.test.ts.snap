// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snowpack + React + Ts 1`] = `
Object {
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"snowpack dev\\",
    \\"build\\": \\"snowpack build\\"
  },
  \\"dependencies\\": {
    \\"react\\": \\"^1.0.0\\",
    \\"react-dom\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@types/react\\": \\"^1.0.0\\",
    \\"@types/react-dom\\": \\"^1.0.0\\",
    \\"snowpack\\": \\"^1.0.0\\"
  }
}",
  "snowpack.config.json": "{
  \\"mount\\": {
    \\"dist\\": \\"/\\",
    \\"src\\": \\"/\\"
  }
}",
  "src": Object {
    "App.tsx": "import * as React from 'react'

interface Props {
  name: string
}

const App = (props: Props) => (
  <>
    <h1>Hello {props.name}</h1>
  </>
)

export default App",
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>intt-app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script type=\\"module\\" src=\\"index.js\\"></script>
  </body>
</html>",
    "index.tsx": "import * as React from 'react'
import * as ReactDOM from 'react-dom'
import App from './App'

const mountNode = document.getElementById('app')
ReactDOM.render(<App name=\\"Jane\\" />, mountNode)",
  },
  "tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"outDir\\": \\"./dist/\\",
    \\"sourceMap\\": true,
    \\"strict\\": true,
    \\"noImplicitReturns\\": true,
    \\"noImplicitAny\\": true,
    \\"module\\": \\"es6\\",
    \\"moduleResolution\\": \\"node\\",
    \\"target\\": \\"es5\\",
    \\"allowJs\\": true,
    \\"jsx\\": \\"react\\"
  },
  \\"include\\": [
    \\"./src/**/*\\"
  ]
}",
}
`;

exports[`Snowpack + React 1`] = `
Object {
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"snowpack dev\\",
    \\"build\\": \\"snowpack build\\"
  },
  \\"dependencies\\": {
    \\"react\\": \\"^1.0.0\\",
    \\"react-dom\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"snowpack\\": \\"^1.0.0\\"
  }
}",
  "snowpack.config.json": "{
  \\"mount\\": {
    \\"dist\\": \\"/\\",
    \\"src\\": \\"/\\"
  }
}",
  "src": Object {
    "App.jsx": "import React from 'react'

const App = props => (
  <>
    <h1>Hello {props.name}</h1>
  </>
)

export default App",
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>intt-app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script type=\\"module\\" src=\\"index.js\\"></script>
  </body>
</html>",
    "index.jsx": "import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'

var mountNode = document.getElementById('app')
ReactDOM.render(<App name=\\"Jane\\" />, mountNode)",
  },
}
`;

exports[`Snowpack + Vue + Ts 1`] = `
Object {
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"snowpack dev\\",
    \\"build\\": \\"snowpack build\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@snowpack/plugin-vue\\": \\"^1.0.0\\",
    \\"snowpack\\": \\"^1.0.0\\"
  }
}",
  "snowpack.config.json": "{
  \\"mount\\": {
    \\"dist\\": \\"/\\",
    \\"src\\": \\"/\\"
  },
  \\"plugins\\": [
    \\"@snowpack/plugin-vue\\"
  ]
}",
  "src": Object {
    "App.vue": "<template>
  <div>
    <h1>{{ name }}</h1>
  </div>
</template>

<script lang=\\"ts\\">
export default {
  setup() {
    return { name: 'Hello World!' }
  }
}
</script>",
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>intt-app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script type=\\"module\\" src=\\"index.js\\"></script>
  </body>
</html>",
    "index.ts": "import { createApp } from 'vue'
import App from './App.vue'

createApp(App).mount('#app')",
    "shims-vue.d.ts": "declare module '*.vue' {
  import type { DefineComponent } from 'vue'
  const component: DefineComponent<{}, {}, any>
  export default component
}",
  },
  "tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"outDir\\": \\"./dist/\\",
    \\"sourceMap\\": true,
    \\"strict\\": true,
    \\"noImplicitReturns\\": true,
    \\"noImplicitAny\\": true,
    \\"module\\": \\"es6\\",
    \\"moduleResolution\\": \\"node\\",
    \\"target\\": \\"es5\\",
    \\"allowJs\\": true,
    \\"jsx\\": \\"react\\"
  },
  \\"include\\": [
    \\"./src/**/*\\"
  ]
}",
}
`;

exports[`Snowpack + Vue 1`] = `
Object {
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"snowpack dev\\",
    \\"build\\": \\"snowpack build\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@snowpack/plugin-vue\\": \\"^1.0.0\\",
    \\"snowpack\\": \\"^1.0.0\\"
  }
}",
  "snowpack.config.json": "{
  \\"mount\\": {
    \\"dist\\": \\"/\\",
    \\"src\\": \\"/\\"
  },
  \\"plugins\\": [
    \\"@snowpack/plugin-vue\\"
  ]
}",
  "src": Object {
    "App.vue": "<template>
  <div>
    <h1>{{ name }}</h1>
  </div>
</template>

<script>
export default {
  setup() {
    return { name: 'Hello World!' }
  }
}
</script>",
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>intt-app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script type=\\"module\\" src=\\"index.js\\"></script>
  </body>
</html>",
    "index.js": "import { createApp } from 'vue'
import App from './App.vue'

createApp(App).mount('#app')",
  },
}
`;

exports[`Webpack + React + AVA 1`] = `
Object {
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "ava.config.js": "export default {
  files: ['tests/**/*']
}",
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"snowpack dev\\",
    \\"build\\": \\"snowpack build\\",
    \\"test\\": \\"ava\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@snowpack/plugin-vue\\": \\"^1.0.0\\",
    \\"ava\\": \\"^1.0.0\\",
    \\"snowpack\\": \\"^1.0.0\\"
  }
}",
  "snowpack.config.json": "{
  \\"mount\\": {
    \\"dist\\": \\"/\\",
    \\"src\\": \\"/\\"
  },
  \\"plugins\\": [
    \\"@snowpack/plugin-vue\\"
  ]
}",
  "src": Object {
    "App.vue": "<template>
  <div>
    <h1>{{ name }}</h1>
  </div>
</template>

<script>
export default {
  setup() {
    return { name: 'Hello World!' }
  }
}
</script>",
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>intt-app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script type=\\"module\\" src=\\"index.js\\"></script>
  </body>
</html>",
    "index.js": "import { createApp } from 'vue'
import App from './App.vue'

createApp(App).mount('#app')",
  },
  "tests": Object {
    "index.test.js": "const test = require('ava')

test('foo', t => {
  t.pass()
})

test('bar', async t => {
  const bar = Promise.resolve('bar')
  t.is(await bar, 'bar')
})",
  },
}
`;

exports[`Webpack + React + Babel + Ts 1`] = `
Object {
  ".babelrc": "{
  \\"presets\\": [
    [
      \\"@babel/preset-env\\",
      {
        \\"modules\\": false
      }
    ],
    \\"@babel/preset-react\\"
  ]
}",
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "dist": Object {
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script src=\\"index.js\\"></script>
  </body>
</html>",
  },
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"webpack serve --mode development\\",
    \\"build\\": \\"webpack --mode production\\"
  },
  \\"dependencies\\": {
    \\"react\\": \\"^1.0.0\\",
    \\"react-dom\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@babel/core\\": \\"^1.0.0\\",
    \\"@babel/preset-env\\": \\"^1.0.0\\",
    \\"@babel/preset-react\\": \\"^1.0.0\\",
    \\"babel-loader\\": \\"^1.0.0\\",
    \\"webpack\\": \\"^1.0.0\\",
    \\"webpack-cli\\": \\"^1.0.0\\",
    \\"webpack-dev-server\\": \\"^1.0.0\\"
  }
}",
  "src": Object {
    "App.jsx": "import React from 'react'

const App = props => (
  <>
    <h1>Hello {props.name}</h1>
  </>
)

export default App",
    "index.jsx": "import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'

var mountNode = document.getElementById('app')
ReactDOM.render(<App name=\\"Jane\\" />, mountNode)",
  },
  "webpack.config.js": "const path = require('path')


const config = {
  entry: './src/index.jsx',
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: 'index.js'
  },
  devServer: {
    contentBase: './dist'
  },
  resolve: {
    extensions: [
      '.js',
      '.jsx',
      '.tsx'
    ]
  },
  module: {
    rules: [
      {
        test: /\\\\.js(x)?$/,
        loader: 'babel-loader',
        exclude: /node_modules/
      }
    ]
  },
  plugins: []
}

module.exports = config",
}
`;

exports[`Webpack + React + Babel 1`] = `
Object {
  ".babelrc": "{
  \\"presets\\": [
    [
      \\"@babel/preset-env\\",
      {
        \\"modules\\": false
      }
    ],
    \\"@babel/preset-react\\"
  ]
}",
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "dist": Object {
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script src=\\"index.js\\"></script>
  </body>
</html>",
  },
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"webpack serve --mode development\\",
    \\"build\\": \\"webpack --mode production\\"
  },
  \\"dependencies\\": {
    \\"react\\": \\"^1.0.0\\",
    \\"react-dom\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@babel/core\\": \\"^1.0.0\\",
    \\"@babel/preset-env\\": \\"^1.0.0\\",
    \\"@babel/preset-react\\": \\"^1.0.0\\",
    \\"babel-loader\\": \\"^1.0.0\\",
    \\"webpack\\": \\"^1.0.0\\",
    \\"webpack-cli\\": \\"^1.0.0\\",
    \\"webpack-dev-server\\": \\"^1.0.0\\"
  }
}",
  "src": Object {
    "App.jsx": "import React from 'react'

const App = props => (
  <>
    <h1>Hello {props.name}</h1>
  </>
)

export default App",
    "index.jsx": "import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'

var mountNode = document.getElementById('app')
ReactDOM.render(<App name=\\"Jane\\" />, mountNode)",
  },
  "webpack.config.js": "const path = require('path')


const config = {
  entry: './src/index.jsx',
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: 'index.js'
  },
  devServer: {
    contentBase: './dist'
  },
  resolve: {
    extensions: [
      '.js',
      '.jsx',
      '.tsx'
    ]
  },
  module: {
    rules: [
      {
        test: /\\\\.js(x)?$/,
        loader: 'babel-loader',
        exclude: /node_modules/
      }
    ]
  },
  plugins: []
}

module.exports = config",
}
`;

exports[`Webpack + React + Jest 1`] = `
Object {
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "dist": Object {
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script src=\\"index.js\\"></script>
  </body>
</html>",
  },
  "jest.config.js": "",
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"webpack serve --mode development\\",
    \\"build\\": \\"webpack --mode production\\",
    \\"test\\": \\"jest\\"
  },
  \\"dependencies\\": {
    \\"react\\": \\"^1.0.0\\",
    \\"react-dom\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"jest\\": \\"^1.0.0\\",
    \\"webpack\\": \\"^1.0.0\\",
    \\"webpack-cli\\": \\"^1.0.0\\",
    \\"webpack-dev-server\\": \\"^1.0.0\\"
  }
}",
  "src": Object {
    "App.jsx": "import React from 'react'

const App = props => (
  <>
    <h1>Hello {props.name}</h1>
  </>
)

export default App",
    "index.jsx": "import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'

var mountNode = document.getElementById('app')
ReactDOM.render(<App name=\\"Jane\\" />, mountNode)",
  },
  "tests": Object {
    "index.test.js": "test('adds 1 + 2 to equal 3', () => {
  expect(1 + 2).toBe(3);
});",
  },
  "webpack.config.js": "const path = require('path')


const config = {
  entry: './src/index.jsx',
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: 'index.js'
  },
  devServer: {
    contentBase: './dist'
  },
  resolve: {
    extensions: [
      '.js',
      '.jsx',
      '.tsx'
    ]
  },
  module: {
    rules: []
  },
  plugins: []
}

module.exports = config",
}
`;

exports[`Webpack + Vue + Babel + Ts 1`] = `
Object {
  ".babelrc": "{
  \\"presets\\": [
    [
      \\"@babel/preset-env\\",
      {
        \\"modules\\": false
      }
    ]
  ]
}",
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "dist": Object {
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script src=\\"index.js\\"></script>
  </body>
</html>",
  },
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"webpack serve --mode development\\",
    \\"build\\": \\"webpack --mode production\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@babel/core\\": \\"^1.0.0\\",
    \\"@babel/preset-env\\": \\"^1.0.0\\",
    \\"@vue/compiler-sfc\\": \\"^1.0.0\\",
    \\"babel-loader\\": \\"^1.0.0\\",
    \\"ts-loader\\": \\"^1.0.0\\",
    \\"typescript\\": \\"^1.0.0\\",
    \\"vue-loader\\": \\"^1.0.0\\",
    \\"webpack\\": \\"^1.0.0\\",
    \\"webpack-cli\\": \\"^1.0.0\\",
    \\"webpack-dev-server\\": \\"^1.0.0\\"
  }
}",
  "src": Object {
    "App.vue": "<template>
  <div>
    <h1>{{ name }}</h1>
  </div>
</template>

<script lang=\\"ts\\">
export default {
  setup() {
    return { name: 'Hello World!' }
  }
}
</script>",
    "index.ts": "import { createApp } from 'vue'
import App from './App.vue'

createApp(App).mount('#app')",
    "shims-vue.d.ts": "declare module '*.vue' {
  import type { DefineComponent } from 'vue'
  const component: DefineComponent<{}, {}, any>
  export default component
}",
  },
  "tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"outDir\\": \\"./dist/\\",
    \\"sourceMap\\": true,
    \\"strict\\": true,
    \\"noImplicitReturns\\": true,
    \\"noImplicitAny\\": true,
    \\"module\\": \\"es6\\",
    \\"moduleResolution\\": \\"node\\",
    \\"target\\": \\"es5\\",
    \\"allowJs\\": true,
    \\"jsx\\": \\"react\\"
  },
  \\"include\\": [
    \\"./src/**/*\\"
  ]
}",
  "webpack.config.js": "const path = require('path')
const { VueLoaderPlugin } = require('vue-loader')

const config = {
  entry: './src/index.ts',
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: 'index.js'
  },
  devServer: {
    contentBase: './dist'
  },
  resolve: {
    extensions: [
      '.js',
      '.ts',
      '.vue'
    ]
  },
  module: {
    rules: [
      {
        test: /\\\\.ts(x)?$/,
        loader: 'ts-loader',
        options: {
          appendTsSuffixTo: [
            /\\\\.vue$/
          ]
        },
        exclude: /node_modules/
      },
      {
        test: /\\\\.vue$/,
        loader: 'vue-loader',
        exclude: /node_modules/
      },
      {
        test: /\\\\.js(x)?$/,
        loader: 'babel-loader',
        exclude: /node_modules/
      }
    ]
  },
  plugins: [
    new VueLoaderPlugin()
  ]
}

module.exports = config",
}
`;

exports[`Webpack + Vue 1`] = `
Object {
  ".gitignore": ".cache/
coverage/
dist/*
!dist/index.html
node_modules/
*.log

# OS generated files
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db",
  "dist": Object {
    "index.html": "<!DOCTYPE html>
<html>
  <head>
    <title>app</title>
    <meta charset=\\"utf-8\\" />
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script src=\\"index.js\\"></script>
  </body>
</html>",
  },
  "package.json": "{
  \\"name\\": \\"intt-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.js\\",
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"start\\": \\"webpack serve --mode development\\",
    \\"build\\": \\"webpack --mode production\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^1.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@vue/compiler-sfc\\": \\"^1.0.0\\",
    \\"vue-loader\\": \\"^1.0.0\\",
    \\"webpack\\": \\"^1.0.0\\",
    \\"webpack-cli\\": \\"^1.0.0\\",
    \\"webpack-dev-server\\": \\"^1.0.0\\"
  }
}",
  "src": Object {
    "App.vue": "<template>
  <div>
    <h1>{{ name }}</h1>
  </div>
</template>

<script>
export default {
  setup() {
    return { name: 'Hello World!' }
  }
}
</script>",
    "index.js": "import { createApp } from 'vue'
import App from './App.vue'

createApp(App).mount('#app')",
  },
  "webpack.config.js": "const path = require('path')
const { VueLoaderPlugin } = require('vue-loader')

const config = {
  entry: './src/index.js',
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: 'index.js'
  },
  devServer: {
    contentBase: './dist'
  },
  resolve: {
    extensions: [
      '.js',
      '.vue',
      '.ts'
    ]
  },
  module: {
    rules: [
      {
        test: /\\\\.vue$/,
        loader: 'vue-loader',
        exclude: /node_modules/
      }
    ]
  },
  plugins: [
    new VueLoaderPlugin()
  ]
}

module.exports = config",
}
`;
